pkg/in/ratelimit.go

```diff
@@ -0,0 +1,29 @@
+package in
+
+import (
+	"github.com/bookingcom/nanotube/pkg/ratelimiter"
+	"time"
+)
+
+func rateLimit(rateLimiters []ratelimiter.RateLimiter, n int, retryDuration time.Duration) {
+	for _, rl := range rateLimiters {
+		for !rl.Allow(int64(n)) {
+			time.Sleep(retryDuration)
+		}
+	}
+}
+
+func newRateLimiterTicker(intervalDuration time.Duration) (c <-chan time.Time, stop func()) {
+	if intervalDuration <= 0 {
```

Comment:
This check seems redundant. Also, check for a negative duration is rather weird.
If we try to do defensive programming here, it would make much more sense to return an error here or something similar. But, honestly, I don't see a reason to have this check.

Reply:
If we remove that check, this seems like a proxy function. I'd consider removing it.